### DISCUSSION TITLE
Disabling zrange contribution to the EKF with the Flow deck

### DISCUSSION THREAD

**Discussion by fehmi8** [bitcraze/discussions#709] [Q&A]
Hi there,

I'm trying to get the crazyflie to not crash when it's flying over an obstacle or when I wave my hand under it. It's using the Flow deck v2.

I commented those lines in the firmware (2023.02): 
https://github.com/bitcraze/crazyflie-firmware/blob/cbbbbdee7ad6fd4cf9d50dff722a7693bee181b5/src/deck/drivers/src/zranger2.c#L140-L143

But now the crazyflie doesn't take off: propellers go crazy but it desperately stays stuck on the ground 😭

Any better idea how I can achieve that? Thanks!

**Comment by knmcguire** [comment]
So currently what is happening, is that it lost all the measurements to determined it's height and maybethis is based on Baro now. What you now see happening, that you might want to make it take off to 1 meter, but currently it thinks that it is at a higher altitude than that, so it is trying desperately to go down. You can check this by looking at the state estimate either on the [flight tab](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/flightcontrol_tab/) or look at the [stateEstimate.z](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#stateestimatez) log on the [plotter tab.](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/plotter_tab/)

Instead of removing all height measurements going into the EKF that are necessary for height estimation, you should either think about adding some kind of step detection (not letting the measurements through if it detects something) or an outlier filter like a low pass filter or such. We don't have that implemented ourselves but some people has requested for this on [the old forum](https://forum.bitcraze.io/viewtopic.php?p=21528&hilit=flying+over#p21528) and here was a [blogpost about fusing it with the multiranger up ranger](https://www.bitcraze.io/2022/01/a-modified-ekf-for-constant-altitude-flights/). We also have some outlier filter examples that have been implemented for the lighthouse and loco positioning systems here: https://github.com/bitcraze/crazyflie-firmware/tree/master/src/modules/src/outlierfilter

**Reply by knmcguire** [reply]
This issue is also somewhat related: https://github.com/bitcraze/crazyflie-firmware/issues/540
